api/features/features_service.py

```diff
@@ -1,33 +1,107 @@
 import typing
+from concurrent.futures import ThreadPoolExecutor
 
+from edge_api.identities.edge_identity_service import (
+    get_edge_identity_overrides,
+)
 from features.dataclasses import EnvironmentFeatureOverridesData
 from features.versioning.versioning_service import get_environment_flags_list
+from projects.models import IdentityOverridesV2MigrationStatus, Project
 
 if typing.TYPE_CHECKING:
     from environments.models import Environment
 
 
+OverridesData = dict[int, EnvironmentFeatureOverridesData]
+
+
 def get_overrides_data(
+    project: Project,
```

Comment:
I'm not entirely sure why we need to pass in the project here rather than just using `environment.project`?

Reply:
Fixed.